src/controllers/auth.controller.ts

```diff
@@ -67,4 +68,54 @@ export const getHospitalDetail = async (req: Request, res: Response) => {
     } catch (error) {
         return res.status(500).json({ success: false, message: 'Error occurred when fetching data!' })
     }
+}
+
+export const registerPatient = async (req: Request, res: Response) => {
+    let data: patientIn = req.body;
+
+    try {
+        const error = checkValidity(data, patientRegRule);
+        if(error) {
+            return res.status(400).json({
+                success: false, message: error
+            })
+        }
+        data.password = await hashPassword(data.password);
```

Comment:
registering a patient does not require password. Hospitals registers them just with their details